Model and JSON For Shift Request

EzekielNicholas123 opened the pull request:
## Describe your changes

## Issue ticket number and link

EzekielNicholas123 committed:
Model and JSON For Shift Request

emilymclean commented on fireapp/lib/domain/models/shift_request.dart:
```diff
@@ -0,0 +1,35 @@
+}
+
+@freezed
+class Volunteer with _$Volunteer {
```
This is not a good name, please name it ShiftVolunteer

emilymclean commented on fireapp/lib/domain/models/shift_request.dart:
```diff
@@ -0,0 +1,35 @@
+    required int volunteerId,
+    required String volunteerGivenName,
+    required String volunteerSurname,
+    required String mobileNumber,
```
For some reason the backend calls this field "mobile_number" (which breaks the pattern of every other field), please add a JsonKey annotation

emilymclean commented on fireapp/mock-data/shift_request.json:
```diff
@@ -0,0 +1,47 @@
+{
+  "results": [
```
Your models can't currently parse this, could you please create a generic type parameter model to hold results? It would be called Results, accept a single type parameter, and have a single field called "Results" of that type parameter.

EzekielNicholas123 replied:
How do I do that?

emilymclean replied:
```dart
class Result<T> {
  
}
```

with a field of `T result`. Do the whole freezed thing. If that doesn't compile tell me

EzekielNicholas123 replied:
@freezed
class Results<T> with _$Results<T> {
  const factory Results({
    @JsonKey(name: "results") required List<T> results,
  }) = _Results<T>;

  factory Results.fromJson(Map<String, Object?> json, T Function(Object? json) fromJsonT) => _$ResultsFromJson(json, fromJsonT);
}

EzekielNicholas123 replied:
but it wont compile

emilymclean reviewed (changes requested):
This is looking good, just a couple minor changes and this will be ready to merge

EzekielNicholas123 committed:
Update shift_request.dart

Fix some bugs

Commit:
Add Results model

- Added a clean task to the build.gradle file in the android directory.
- Created a new Results model in the domain/models/results.dart file. This model represents a list of results.
- Generated code for the Results model using freezed_annotation and json_serializable.

fix: Update ShiftVolunteer model in shift_request.dart

- Updated the ShiftVolunteer model in the shift_request.dart file.
- Added a required annotation for the mobileNumber field.

chore: Update GeneratedPluginRegistrant.swift and pubspec.yaml

- Updated imports in GeneratedPluginRegistrant.swift to use path_provider_foundation and shared_preferences_foundation instead of their macOS counterparts.
- Updated dev_dependencies in pubspec.yaml to use build_runner version 2.4.6 instead of an older version.
